### PR TITLE
Prometheus fixes post unified source ingestion framework

### PR DESCRIPTION
**Pull request by nacrooks**
This PR changes prometheus metrics to not be specific to Kafka.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/3539)

### CODE REVIEW

File: src/dataflow/src/source/kafka.rs [hunk 1]
```diff
@@ -31,6 +37,35 @@ use crate::server::{
...
+    )
+    .unwrap();
+    static ref KAFKA_MESSAGES_INGESTED: IntCounterVec = register_int_counter_vec!(
+        "mz_kafka_messages_ingested",
+        "The number of messages ingested per partition.",
+        &["topic", "source_id", "partition_id"]
+    )
+    .unwrap();
+}
+
```
Comment (quodlibetor) [inline src/dataflow/src/source/kafka.rs:68]: We don't need these, sorry about the miscommunication.

File: src/dataflow/src/source/mod.rs [hunk 2]
```diff
@@ -556,8 +564,8 @@ impl SourceMetrics {
             )
             .unwrap();
             static ref CAPABILITY: UIntGaugeVec = register_uint_gauge_vec!(
-                "mz_kafka_capability",
-                "The current capability for this dataflow. This corresponds to min(mz_kafka_partition_closed_ts)",
+                "mz_capability",
+                "The current capability for this dataflow. This corresponds to min(mz_partition_closed_ts)",
                 &["topic", "source_id", "worker_id"]
```
Comment (quodlibetor) [inline src/dataflow/src/source/mod.rs:569]: Does `topic` make sense for every kind of source? I think, but am open to other ideas, that it makes more sense to leave this as `mz_kafka_capability` and add other metrics that have more specific labels (e.g. kinesis, file, etc.

File: src/dataflow/src/source/mod.rs [hunk 3]
```diff
@@ -556,8 +564,8 @@ impl SourceMetrics {
...
             )
             .unwrap();
```
Comment (quodlibetor) [inline src/dataflow/src/source/mod.rs:571]: (I can't comment outside the range of the change context)

Similarly, below on lines 590 to 609, it seems like either we should have different kinds of metrics depending on the source, or at least an extra `"source_type"` label that can be e.g. `"file"`, `"kafka"`, `"kinesis"`

**Comment by nacrooks** [comment]
@quodlibetor : this PR was just to get the configs working again, I was hoping to do what you suggested in a separate PR. Do you mind expanding on what you mean by "It might be worth splitting the dashboard json config updates into a separate PR so that we can get them working again while we think through the metrics?"?

**Comment by quodlibetor** [comment]
> Do you mind expanding on what you mean

IIUC think that this PR could/should remove the legacy metrics addition https://github.com/MaterializeInc/materialize/pull/3539/files#diff-00e9971b77083831d879b28eac308d85R40
And I'm not sure if [the `mz_kafka_capability` -> `mz_capability` change](https://github.com/MaterializeInc/materialize/pull/3539/files#r450917051) is required to get things working again or should only go in once the labels are sorted out.

**Comment by nacrooks** [comment]
I'll remove the legacy metric addition for sure. Re the mz_capability renaming: it just felt wrong to have something mz_kafka_* in source/mod.rs as this code is not Kafka specific. Happy to leave it in though.

**Comment by quodlibetor** [comment]
> mz_capability

I think it'll be useful when you rework this to have `mz_capability` also have a `source_type` label, but otherwise I agree with the sentiment that there probably shouldn't be things named `*kafka*` in that module.

**Comment by nacrooks** [comment]
Ok. Will merged this and created issue: https://github.com/MaterializeInc/materialize/issues/3540

**Pull request merged**